ngraph_bridge/enable_variable_ops/ngraph_variable_update_ng_tensor_op.cc

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use thi0s file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "tensorflow/core/common_runtime/dma_helper.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/resource_mgr.h"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/lib/strings/strcat.h"
+#include "tensorflow/core/platform/default/logging.h"
+
+#include "ngraph/event_tracing.hpp"
+
+#include "ngraph_bridge/enable_variable_ops/ngraph_var.h"
+#include "ngraph_bridge/enable_variable_ops/ngraph_variable_update_ng_tensor_op.h"
+#include "ngraph_bridge/ngraph_timer.h"
+#include "ngraph_bridge/ngraph_utils.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+//---------------------------------------------------------------------------
+//  NGraphVariableUpdateNGTensorOp::ctor
+//---------------------------------------------------------------------------
+NGraphVariableUpdateNGTensorOp::NGraphVariableUpdateNGTensorOp(
+    OpKernelConstruction* context)
+    : OpKernel(context) {
+  OP_REQUIRES_OK(context, context->GetAttr("ngraph_graph_id", &ng_graph_id_));
+  OP_REQUIRES_OK(context, context->GetAttr("ngraph_variable_shared_name",
+                                           &ng_variable_shared_name_));
+
+  NGRAPH_VLOG(4) << "NGraphVariableUpdateNGTensorOp:: Constructor called for: "
+                 << def().name() << " ,Graph ID " << ng_graph_id_
+                 << " ngraph_variable_shared_name " << ng_variable_shared_name_;
+
+  OP_REQUIRES(context, IsRefType(context->input_type(0)),
+              errors::InvalidArgument("lhs input needs to be a ref type"));
```

Comment:
There is only one input. Reword to
errors::InvalidArgument("Input[0] needs to be a ref type"));